runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -133,24 +158,57 @@ MM_IndexableObjectAllocationModel::initializeIndexableObject(MM_EnvironmentBase
 #endif /* defined(J9VM_ENV_DATA64) */
 		} else {
 			indexableObjectModel->setSizeInElementsForContiguous(spine, _numberOfIndexedFields);
+			isArrayletDataAdjacentToHeader = indexableObjectModel->isArrayletDataAdjacentToHeader(spine);
+			if (isArrayletDataAdjacentToHeader) {
+				getAllocateDescription()->setDataAdjacentToHeader(true);
 #if defined(J9VM_ENV_DATA64)
 			if (((J9JavaVM *)env->getLanguageVM())->isIndexableDataAddrPresent) {
 				indexableObjectModel->setDataAddrForContiguous(spine);
 			}
 #endif /* defined(J9VM_ENV_DATA64) */
+			} else if (isAllIndexableDataContiguousEnabled) {
```

Comment:
@amicic Do we need (conceptually) expose this check for all platforms? If not we can join sections above and below to one to make code cleaner